packages/core/src/model.js

```diff
@@ -1419,6 +1419,23 @@ ${associationOwner._getAssociationDebugList()}`);
         // TODO: support composite primary keys
         [this.primaryKeyAttribute]: param,
       };
+    } else if (typeof param === 'object') {
+      // composite primary key support
+      options.where = Object.values(this.primaryKeys).reduce((where, pkMetadata) => {
```

Comment:
Nit: I'm not a huge fan of `Array.prototype.reduce`, so would always prefer a `for` loop. That being said, it is used in a few places across the library 🤔 

This is hot code though, and it looks like this construct creates a lot of temporary objects when the primary key has many columns, so I think it'd be better to just build up a single object in a `for` loop.

Reply:
yeah i tried to get a feel for the style of the library, but you're right that its hot code and should create the least tmp stuff as possible